crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -0,0 +1,528 @@
+    NoDir = 0x80100025,
+    /// SCARD_E_READER_UNSUPPORTED
+    ReaderUnsupported = 0x8010001A,
+    /// SCARD_E_DUPLICATE_READER
+    DuplicateReader = 0x8010001B,
+    /// SCARD_E_CARD_UNSUPPORTED
+    CardUnsupported = 0x8010001C,
+    /// SCARD_E_NO_SERVICE
+    NoService = 0x8010001D,
+    /// SCARD_E_SERVICE_STOPPED
+    ServiceStopped = 0x8010001E,
+    /// SCARD_E_UNEXPECTED
+    Unexpected = 0x8010001F,
+    /// SCARD_E_NO_FILE
+    NoFile = 0x80100026,
+    /// SCARD_E_NO_ACCESS
+    NoAccess = 0x80100027,
+    /// SCARD_E_WRITE_TOO_MANY
+    WriteTooMany = 0x80100028,
+    /// SCARD_E_BAD_SEEK
+    BadSeek = 0x80100029,
+    /// SCARD_E_INVALID_CHV
+    InvalidChv = 0x8010002A,
+    /// SCARD_E_UNKNOWN_RES_MSG
+    UnknownResMsg = 0x8010002B,
+    /// SCARD_E_NO_SUCH_CERTIFICATE
+    NoSuchCertificate = 0x8010002C,
+    /// SCARD_E_CERTIFICATE_UNAVAILABLE
+    CertificateUnavailable = 0x8010002D,
+    /// SCARD_E_NO_READERS_AVAILABLE
+    NoReadersAvailable = 0x8010002E,
+    /// SCARD_E_COMM_DATA_LOST
+    CommDataLost = 0x8010002F,
+    /// SCARD_E_NO_KEY_CONTAINER
+    NoKeyContainer = 0x80100030,
+    /// SCARD_E_SERVER_TOO_BUSY
+    ServerTooBusy = 0x80100031,
+    /// SCARD_E_PIN_CACHE_EXPIRED
+    PinCacheExpired = 0x80100032,
+    /// SCARD_E_NO_PIN_CACHE
+    NoPinCache = 0x80100033,
+    /// SCARD_E_READ_ONLY_CARD
+    ReadOnlyCard = 0x80100034,
+    /// SCARD_W_UNSUPPORTED_CARD
+    UnsupportedCard = 0x80100065,
+    /// SCARD_W_UNRESPONSIVE_CARD
+    UnresponsiveCard = 0x80100066,
+    /// SCARD_W_UNPOWERED_CARD
+    UnpoweredCard = 0x80100067,
+    /// SCARD_W_RESET_CARD
+    ResetCard = 0x80100068,
+    /// SCARD_W_REMOVED_CARD
+    RemovedCard = 0x80100069,
+    /// SCARD_W_SECURITY_VIOLATION
+    SecurityViolation = 0x8010006A,
+    /// SCARD_W_WRONG_CHV
+    WrongChv = 0x8010006B,
+    /// SCARD_W_CHV_BLOCKED
+    ChvBlocked = 0x8010006C,
+    /// SCARD_W_EOF
+    Eof = 0x8010006D,
+    /// SCARD_W_CANCELLED_BY_USER
+    CancelledByUser = 0x8010006E,
+    /// SCARD_W_CARD_NOT_AUTHENTICATED
+    CardNotAuthenticated = 0x8010006F,
+    /// SCARD_W_CACHE_ITEM_NOT_FOUND
+    CacheItemNotFound = 0x80100070,
+    /// SCARD_W_CACHE_ITEM_STALE
+    CacheItemStale = 0x80100071,
+    /// SCARD_W_CACHE_ITEM_TOO_BIG
+    CacheItemTooBig = 0x80100072,
+}
+
+pub mod rpce {
+    //! PDUs for [\[MS-RPCE\]: Remote Procedure Call Protocol Extensions] as required by [MS-RDPESC].
+    //!
+    //! [\[MS-RPCE\]: Remote Procedure Call Protocol Extensions]: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rpce/290c38b1-92fe-4229-91e6-4fc376610c15
+
+    use ironrdp_pdu::{cursor::WriteCursor, ensure_size, invalid_message_err, PduError, PduResult};
+
+    /// Trait for types that can be encoded into an [MS-RPCE] message.
+    ///
+    /// Implementers should typically avoid implementing this trait directly
+    /// and instead implement [`HeaderlessEncode`].
+    pub trait Encode: Send + Sync + std::fmt::Debug {
```

Comment:
Note that you may declare `Encode` here as a super trait of `PduEncode` which requires `std::fmt::Debug`
```suggestion
    pub trait Encode: PduEncode + std::fmt::Debug {
```

Reply:
Having removed the constraints on `PduEncode`, I went with this: a892f4eca2267db91787c6ede8c589cdae873b19